diff.go

```diff
@@ -0,0 +1,208 @@
+package godump
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"text/tabwriter"
+)
+
+// Diff prints a diff between two values to stdout.
+func Diff(a, b any) {
+	defaultDumper.Diff(a, b)
+}
+
+// Diff prints a diff between two values to the configured writer.
+func (d *Dumper) Diff(a, b any) {
+	fmt.Fprint(d.writer, d.DiffStr(a, b))
+}
+
+// DiffStr returns a string diff between two values.
+func DiffStr(a, b any) string {
+	return defaultDumper.DiffStr(a, b)
+}
+
+// DiffStr returns a string diff between two values.
+func (d *Dumper) DiffStr(a, b any) string {
+	var sb strings.Builder
+	d.printDiffHeader(&sb)
+	d.ensureColorizer()
+
+	dumps := d.diffDumps(a, b)
+	leftLines := splitLines(dumps.left)
+	rightLines := splitLines(dumps.right)
+	ops := diffLines(leftLines, rightLines)
```

Comment:
I have a problem with the way splitLines works.

Consider a diff like this

```diff
-
-
-
+foobar
```

When we diff with 3 \n being replaced by foobar. The diff won't be the one expected

Also, consider `\r\n` existence.

Right now, `\r\n\r\n` is considered differently than  `\n\n`

I would detect the presence of `\r\n` in splitLines to split against it if present